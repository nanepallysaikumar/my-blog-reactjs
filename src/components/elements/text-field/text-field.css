.text-field {
    width: 100%;
    padding: 6px 8px;
    font-size: 18px;
    box-sizing: border-box;
}

.text-field.error {
    border: 1px solid #d64161;
}