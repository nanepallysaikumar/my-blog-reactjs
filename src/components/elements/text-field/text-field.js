import React from 'react';
import classnames from 'classnames';
import { FormField } from '../form-field/form-field';
import './text-field.css';

export const TextField = props => (
    <FormField
        title={props.title}
        name={props.field.name}
        error={props.error}
        touched={props.touched}
    >
        <input
            placeholder={props.placeholder}
            type={props.type || 'text'}
            {...props.field}
            className={classnames('text-field', {
                error: props.error && props.touched
            })}
        />
    </FormField>
);
