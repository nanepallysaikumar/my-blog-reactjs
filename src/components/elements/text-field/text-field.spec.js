import { render, cleanup } from 'react-testing-library';
import React from 'react';
import { TextField } from './text-field';

const setup = (touched, error) => {
    return render(
        <TextField
            title="Test Title"
            field={{
                name: 'test',
                value: 'test',
                onChange: jest.fn()
            }}
            touched={touched}
            error={error}
        />
    );
};

describe('text-field', () => {
    it('should render normal text-field', async () => {
        const { container } = setup(false, undefined);
        expect(container).toMatchSnapshot('text-field-normal-snapshot');
    });

    it('should render error text-field', async () => {
        const { container } = setup(true, 'Invalid');
        expect(container).toMatchSnapshot('text-field-error-snapshot');
    });

    afterEach(cleanup);
});
