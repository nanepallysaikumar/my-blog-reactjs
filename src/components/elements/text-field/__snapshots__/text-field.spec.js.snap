// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text-field should render error text-field: text-field-error-snapshot 1`] = `
<div>
  <div
    class="form-field"
  >
    <label
      class="form-field-label"
    >
      Test Title
    </label>
    <input
      class="text-field error"
      name="test"
      type="text"
      value="test"
    />
    <div
      class="form-field-error"
      data-testid="form-field-error-test"
    >
      Invalid
    </div>
  </div>
</div>
`;

exports[`text-field should render normal text-field: text-field-normal-snapshot 1`] = `
<div>
  <div
    class="form-field"
  >
    <label
      class="form-field-label"
    >
      Test Title
    </label>
    <input
      class="text-field"
      name="test"
      type="text"
      value="test"
    />
  </div>
</div>
`;
